pkg/reconciler/v1alpha1/trigger/trigger.go

```diff
@@ -137,27 +170,31 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	}
 
 	if err != nil {
-		logging.FromContext(ctx).Error("Could not Get Trigger", zap.Error(err))
+		logging.FromContext(ctx).Error("Could not get Trigger", zap.Error(err))
 		return reconcile.Result{}, err
 	}
 
+	// Modify a copy, not the original.
+	trigger = trigger.DeepCopy()
+
 	// Reconcile this copy of the Trigger and then write back any status updates regardless of
 	// whether the reconcile error out.
 	reconcileErr := r.reconcile(ctx, trigger)
 	if reconcileErr != nil {
 		logging.FromContext(ctx).Error("Error reconciling Trigger", zap.Error(reconcileErr))
+		r.recorder.Eventf(trigger, corev1.EventTypeWarning, triggerReconcileFailed, "Trigger reconciliation failed: %v", reconcileErr)
 	} else {
 		logging.FromContext(ctx).Debug("Trigger reconciled")
-		r.recorder.Event(trigger, corev1.EventTypeNormal, triggerReconciled, "Trigger reconciled")
+		r.recorder.Eventf(trigger, corev1.EventTypeNormal, triggerReconciled, "Trigger reconciled: %q", trigger.Name)
```

Comment:
Is the name already there in the sense that the source of the event is the trigger?

Reply:
I added it so that an event line has all the info.
Example:
Normal   TriggerReconciled       2s (x15 over 6h32m)   trigger-controller  Trigger reconciled: "t"

I can remove the name if you prefer

Reply:
How did you get that event? If I describe the Trigger, then it implicitly means that the event ties to that Trigger (so the Trigger's name is redundant, as the name is identical to the source). If I list events, then I can look at the source field, which has the trigger's name already.

Reply:
Yes, you are right... I saw that you merged it in... We won't add that info then in the future, is redundant.